### PR TITLE
 drivers: espi: xec: Allow to intercept eSPI host suspend/reset warnings into the app

### LINKED ISSUE #22188: drivers: espi: xec : eSPI driver should not send VWire SUS_ACK automatically in all cases
**Describe the bug**
When host sends indication of suspend or warm reset warning, eSPI slave should acknowledge that slave system is ready to be suspended or going through warm reset. However, currently eSPI driver automatically sends a virtual wire packet (SUS_ACK) without given opportunity to some apps to perform activities before warm reset or suspend.

However, note some systems benefit from automatically acknowledge

**To Reproduce**
1. mkdir build; cd build
2. cmake -DBOARD=mec1501modular_assy6885 -DZEPHYR_MODULES=/home/jamezaar/ecwork/modules/hal/microchip ..
3. make

4. west flash or flash using dediprog

**Expected behavior**
eSPI slave systems can perform suspend/reset activities when eSPI host is to be suspended or reset.

**Impact**
This may result in system crash or unexpected behavior since there is no gracefully suspend

### PR DESCRIPTION
**Pull request by albertofloyd**
Fix #22188
    Introduce config option to enable automatic eSPI host ackwnowledges
    via virtual wires on response to eSPI host suspend and reset warnings.
    This will give the opportunity for systems that require to perform
    any preparations before reset or suspend.

### CODE REVIEW

File: drivers/espi/Kconfig [hunk 1]
```diff
@@ -22,8 +22,23 @@ config ESPI_SLAVE
 	help
 	  Enables eSPI driver in slave mode.
 
+config ESPI_AUTOMATIC_WARNING_ACKNOWLEDGE
+	bool "ESPI slave driver"
+	default y
+	help
+	  Enable automatic acknowledge from eSPI slave towards eSPI host
+	  whenever it receives suspend or reset warning.
+	  If this is disabled, it means the app wants to be give the oportunity
```
Comment (ulfalizer) [inline drivers/espi/Kconfig]: *opportunity

File: drivers/espi/Kconfig [hunk 2]
```diff
@@ -22,8 +22,23 @@ config ESPI_SLAVE
...
+	  to prepare for either HOST suspend or reset.
+
 config ESPI_INIT_PRIORITY
-	int "IRQ Priority for ESPI Controller"
+	int "Driver initialization priority for ESPI Controller"
+	default 3
+	help
+	  IRQ Priority for ESPI Controller.
+
+config ESPI_IRQ_PRIORITY
```
Comment (franciscomunoz) [inline drivers/espi/Kconfig]: This should come from a DT macro. Remove the entry from Kconfig

interrupts = <11 0>, <15 0>, <7 0>;

The second parameter is the priority

And this are the autogenerated macros
#define DT_MICROCHIP_XEC_ESPI_400F3400_IRQ_0_PRIORITY 0

#define DT_MICROCHIP_XEC_ESPI_400F3400_IRQ_1_PRIORITY 0 

#define DT_MICROCHIP_XEC_ESPI_400F3400_IRQ_2_PRIORITY 0

You can add an entries in the fixup to hide the address

DT_ESPI_MCHP_XEC_0_IRQ_PRIORITY
DT_ESPI_MCHP_XEC_1_IRQ_PRIORITY
DT_ESPI_MCHP_XEC_2_IRQ_PRIORITY

**Review by franciscomunoz — CHANGES_REQUESTED** [review]
Separate the change for the IRQ in a new PR

**Comment by andrewboie** [comment]
> Separate the change for the IRQ in a new PR

I don't think it needs to be in a separate pull request, but it should at least be an independent patch in the series.

**Review by nashif — APPROVED** [review]

**Review by franciscomunoz — APPROVED** [review]

**Pull request merged**